views/providers.py

```diff
@@ -205,9 +218,25 @@ def clear_all_providers(self):
         assert ui_utils(self.web_session).waitForTextOnPage(
             "Delete initiated for 1 Middleware Provider from the CFME Database", 15)
 
-        while not ui_utils(self.web_session).isElementPresent(By.XPATH, "//span[contains(.,'(Item 0 of 0)')]"):
+        count = 0
+        while not ui_utils(self.web_session).isElementPresent(By.XPATH,
+                                                              "//span[contains(.,'(Item 0 of 0)')]") and count < 10:
             self.web_driver.refresh()
+            count += 1
         assert ui_utils(self.web_session).waitForTextOnPage("No Records Found", 15)
         self.web_session.logger.info("All the middleware providers are deleted successfully.")
 
+    def verify_refresh_status_success(self):
+
+        count = 0
+        while ui_utils(self.web_session).isTextOnPage("Never") and count < 10:
+            self.web_driver.refresh()
+            count += 1
+
+        if ui_utils(self.web_session).isTextOnPage("Success"):
+            self.web_session.logger.info("The last refresh status is: Success ")
+            return True
+        else:
+            return False
+
 
```

Comment:
Would be more precise to read the info in python dictionary format, and then evaluate the specific value.
Something like (once get_generic_table_as_dict() is merged):
 page_details = self.ui_utils(web_session).get_generic_table_as_dict()
 refresh_value =  print page_details.get("Last Refresh")